webui/react/src/stores/users.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { createContext, PropsWithChildren, useCallback, useContext, useState } from 'react';
+
+import { globalStorage } from 'globalStorage';
+import { getUsers } from 'services/api';
+import { V1GetUsersRequestSortBy } from 'services/api-ts-sdk';
+import { isEqual } from 'shared/utils/data';
+import { Auth, DetailedUser } from 'types';
+import { getCookie, setCookie } from 'utils/browser';
+import handleError from 'utils/error';
+import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
+
+export type CurrentUser = Auth & { checked: boolean };
+type UsersContext = {
+  auth: Loadable<CurrentUser>;
+  updateAuth: (fn: (users: Loadable<CurrentUser>) => Loadable<CurrentUser>) => void;
+  updateUsers: (fn: (users: Loadable<DetailedUser[]>) => Loadable<DetailedUser[]>) => void;
+  users: Loadable<DetailedUser[]>;
+};
+
+type FetchUsersConfig = {
+  limit: number;
+  offset: number;
+  orderBy: 'ORDER_BY_DESC' | 'ORDER_BY_ASC';
+  sortBy: V1GetUsersRequestSortBy;
+};
+
+type UseUsersReturn = {
```

Comment:
Can we split the data and the functions into separate hooks? `useAuth`, `useUsers`, `useResetAuth`, etc? Once we aren't using react context we can make the actions into non-hooks.